src/defs.h

```diff
@@ -174,6 +182,8 @@ struct var {
     int subscripts_idx;
     rename_t rename;
     ref_block_list_t ref_block_list; /* blocks which kill variable */
+    user_t *users_head;
+    user_t *users_tail;
```

Comment:
It seems like we can access users_tail directly through users_head->prev. Is the users_tail variable still necessary?

Reply:
I misunderstood; I thought you were implementing a circular doubly linked list, but it's actually just a doubly linked list without the circular characteristic.